pkg/queue/sharedmain/handlers.go

```diff
@@ -124,3 +126,15 @@ func adminHandler(ctx context.Context, logger *zap.SugaredLogger, drainer *pkgha
 
 	return mux
 }
+
+func wrapQPHandlerWithFullDuplex(h http.Handler, enableFullDuplex bool, logger *zap.SugaredLogger) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if enableFullDuplex {
+			rc := http.NewResponseController(w)
+			if err := rc.EnableFullDuplex(); err != nil {
+				logger.Errorw("Unable to enable full duplex", zap.Error(err))
+			}
+		}
+		h.ServeHTTP(w, r)
+	})
+}
```

Comment:
nit: we can drop conditional in the request path

```suggestion
func withFullDuplex(h http.Handler, enableFullDuplex bool, logger *zap.SugaredLogger) http.Handler {
	if !enableFullDuplex {
		return h
	}
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {		
		rc := http.NewResponseController(w)
		if err := rc.EnableFullDuplex(); err != nil {
			logger.Errorw("Unable to enable full duplex", zap.Error(err))
		}
		h.ServeHTTP(w, r)
	})
}
```